---
title: "Class 14 RNAseq mini project"
author: "Neva Olliffe PID A69026930"
format: pdf
toc: true
---
# DEseq
## Load libraries
```{r} 
#| message = 0
library(DESeq2)
library(dplyr)
library("AnnotationDbi")
library("org.Hs.eg.db")
library(pathview)
library(gage)
library(gageData)


# In console:
# BiocManager::install( c("pathview", "gage", "gageData") )
```


## Load data
```{r}
colData <- read.csv("GSE37704_metadata.csv", row.names = 1)
countData <- read.csv("GSE37704_featurecounts.csv", row.names = 1)

head(colData )
head(countData)
```

```{r}
# Remove 'length' column
countData <- countData[,-1]
head(countData)
```

## Filter data
Remove genes with 0 reads across all samples. 
```{r}
countData <- countData[rowSums(countData)!=0, ]
head(countData)
```
## Run DESeq
```{r}
dds <- DESeqDataSetFromMatrix(countData = countData,
                              colData = colData, 
                              design=~condition)

dds = DESeq(dds)
```
```{r}
dds
```
```{r}
res = results(dds)
summary(res)
```

## Volcano plot
```{r}
plot(res$log2FoldChange, -log(res$padj))
```

```{r}
# Make a color vector for all genes
mycols <- rep("gray", nrow(res))

# Color red the genes with absolute fold change above 2
mycols[ abs(res$log2FoldChange) > 2 ] <- "red"

# Color blue those with adjusted p-value less than 0.01
#  and absolute fold change more than 2
inds <- (res$padj <0.01) & (abs(res$log2FoldChange) > 2 )
mycols[ inds ] <- "blue"

plot( res$log2FoldChange, -log(res$padj), col=mycols, xlab="Log2(FoldChange)", ylab="-Log(P-value)" )
```

## Add gene annotations and save results
```{r}
columns(org.Hs.eg.db)
gene.Names <- row.names(res)

# Add gene symbols
res$symbol = mapIds(org.Hs.eg.db,
                    keys=gene.Names, 
                    keytype="ENSEMBL",
                    column="SYMBOL",
                    multiVals="first")

# Add entrez IDs
res$entrez = mapIds(org.Hs.eg.db,
                    keys=gene.Names,
                    keytype="ENSEMBL",
                    column="ENTREZID",
                    multiVals="first")

# Add gene names
res$name =   mapIds(org.Hs.eg.db,
                    keys=gene.Names,
                    keytype="ENSEMBL",
                    column="GENENAME",
                    multiVals="first")

head(res, 10)
```
```{r}
# Save results
res = res[order(res$pvalue),]

write.csv(res, file = "deseq_results.csv")
```


# Pathway analysis

## Use Kegg analysis
```{r}
data(kegg.sets.hs)
data(sigmet.idx.hs)

# Focus on signaling and metabolic pathways only
kegg.sets.hs = kegg.sets.hs[sigmet.idx.hs]

# Examine the first 3 pathways
head(kegg.sets.hs, 3)
```
To run gage analysis, we need a vector of fold changes named with Entrez gene ids. We need to create this using `res$entrez` and `res$log2foldchange`.

## Run Gage analysis
```{r}
foldchanges = res$log2FoldChange
names(foldchanges) = res$entrez
head(foldchanges)
```
```{r}
# Run the analysis
keggres = gage(foldchanges, gsets=kegg.sets.hs)

attributes(keggres)
```
```{r}
# Look at the first few down (less) pathways
head(keggres$less)
```
```{r}
# Generate pathway plot for the cell cycle pathway
pathview(gene.data=foldchanges, pathway.id="hsa04110")
```

![](hsa04110.pathview.png)

```{r}
# A different PDF based output of the same data
pathview(gene.data=foldchanges, pathway.id="hsa04110", kegg.native=FALSE)

```

## Focus on the top 5 upregulated pathways. 
```{r}
keggrespathways <- rownames(keggres$greater)[1:5]

# Get the 8-character long ids
keggresids <- substr(keggrespathways, start=1, stop=8)
keggresids

```

```{r}
# Plot all 5 pathways
pathview(gene.data=foldchanges, pathway.id=keggresids, species="hsa")
```
![](hsa04640.pathview.png)
![](hsa04630.pathview.png)
![](hsa00140.pathview.png)
![](hsa04142.pathview.png)
![](hsa04330.pathview.png)

## Get top 5 down-regulated pathways
```{r}
keggrespathways <- rownames(keggres$less)[1:5]

# Get the 8-character long ids
keggresids <- substr(keggrespathways, start=1, stop=8)
keggresids

# Plot all 5 pathways
pathview(gene.data=foldchanges, pathway.id=keggresids, species="hsa")
```

![](hsa04110.pathview.png)
![](hsa03030.pathview.png)
![](hsa03013.pathview.png)
![](hsa03440.pathview.png)
![](hsa04114.pathview.png)

# Gene ontology
```{r}
data(go.sets.hs)
data(go.subs.hs)

# Focus on Biological Process subset of GO
gobpsets = go.sets.hs[go.subs.hs$BP]

gobpres = gage(foldchanges, gsets=gobpsets, same.dir=TRUE)

lapply(gobpres, head)
```

# 4. Reactome analysis
```{r}
# List the significant genes
sig_genes <- res[res$padj <= 0.05 & !is.na(res$padj), "symbol"]
print(paste("Total number of significant genes:", length(sig_genes)))

```
```{r}
# Output the significant genes
write.table(sig_genes, file="significant_genes.txt", row.names=FALSE, col.names=FALSE, quote=FALSE)

```




## Q: What pathway has the most significant “Entities p-value”? Do the most significant pathways listed match your previous KEGG results? What factors could cause differences between the two methods?

"Cell cycle, mitotic" has the most significant entities p-value. Many of the pathway names are not the same, likely because very similar pathways involved in chromosome separation and cell cycle checkpoints are being called slightly differently based on how they are identified in each database.

















